test/cljc/reitit/swagger_test.clj

```diff
@@ -250,7 +282,42 @@
+                                                                        {:$ref "#/definitions/reitit.swagger-test~1req-val"}
+                                                                        :definitions {::req-key
+                                                                                      {:type "string"
+                                                                                       :x-anyOf [{:type "string"}
+                                                                                                 {:type "string"}]}
+                                                                                      ::req-val
+                                                                                      {:type "object"
+                                                                                       :x-anyOf [{:type "object"}
+                                                                                                 {:type "string"}]}}}}]
+                                                         :responses {200
+                                                                     {:schema
+                                                                      {:$ref "#/definitions/reitit.swagger-test~1resp-map"
+                                                                       :x-anyOf [{:$ref "#/definitions/reitit.swagger-test~1resp-map"}
+                                                                                 {:$ref "#/definitions/reitit.swagger-test~1resp-string"}]
+                                                                       :definitions {::resp-map {:type "object"}
+                                                                                     ::resp-string
+                                                                                     {:type "string", :minLength 1}}}
+                                                                      :description ""}
+                                                                     500 {:description "fail"}}
+                                                         :definitions {::req-key {:type "string"
+                                                                                  :x-anyOf [{:type "string"}
+                                                                                            {:type "string"}]}
+                                                                       ::req-val {:type "object"
+                                                                                  :x-anyOf [{:type "object"}
+                                                                                            {:type "string"}]}
+                                                                       ::resp-map {:type "object"}
+                                                                       ::resp-string {:type "string", :minLength 1}}
```

Comment:
So the `:definitions` are now on three levels: inside the `:schema`, inside `:put` and at the top level of the swagger spec. Is that what we want? By reading the spec, it seems only the top-level `:definitions` is supported.

Reply:
Yeah, it's not ideal. Only the top-level one gets used for anything I believe. Want me to explore eliminating the others?

Reply:
If you can find an easy way to do that, it would be nice. I think swagger validators will complain about the intermediate ones.

Reply:
The malli side of that work is in https://github.com/metosin/malli/pull/900